### ISSUE TITLE
Automatically add a job title prefix based on hours in that department + Intern Removal

### ISSUE THREAD

**Issue by keronshb** [space-wizards/space-station-14#22462]
## Description

Instead of having a ton of unique job titles & spawners, the more sane way to approach this is to automatically prefix a title depending on hours in a department.

ex)
- If Hours < 24 in Sci then add "Intern" prefix to Sci job title.
- If hours > 24 in Sci then remove "Intern" prefix to Sci job title
- If hours > 60 in Sci then allow optional "Teacher" prefix to Sci job title

**Comment by BYONDFuckery** [comment]
I don't like this idea, I think intern roles should stay

**Comment by thebadman4662** [comment]
They will stay as title

**Comment by OtherSkittle** [comment]
Based and real?

I would personally add that Intern prefix should never be forced off of people. No matter how big the number of hours you decide to add, there are always going to be people that don't feel comfortable in the role they have selected, plus I don't see how we benefit from not giving people the option to at the very least RP as intern characters. There is also the possibility of players taking a long break from the game and forgetting how a department plays (or it was changed significantly since the last time they played).

Also isn't 24 hours a bit too much? RD at the moment only requires 10 hours of sci gameplay yet with these numbers people can simultaneously be eligible to play the head of the department while also being classified as an intern by the game. Is the intent here to let people play as an Intern Research Director? I don't necessarily disagree with this I just want to clarify.

**Comment by UbaserB** [comment]
OtherSkittle, they basically aren't changing how things work right now, its just a tweak to make it more dynamic without having several different jobs. Though, i do agree with the 24 hours as that is a lot. Either increase the amount of time required to become RD or reduce it to under 10 hours for intern

**Comment by OtherSkittle** [comment]
> OtherSkittle, they basically aren't changing how things work right now, its just a tweak to make it more dynamic without having several different jobs. Though, i do agree with the 24 hours as that is a lot. Either increase the amount of time required to become RD or reduce it to under 10 hours for intern

Yes I know this is basically just an ideas page. All I'm saying is that there should be an option for people who have above 24 hours in a department (or whatever number gets settled on in the end) to toggle the intern prefix, instead of forcing the intern prefix to be removed.

As far as I'm aware the only reason intern roles currently have a department playtime max is because people were stealing newbie roles in order to play in the department they wanted. If this change is implemented then I would assume this would no longer be problem so I do not see why people should ever be locked out advertising themselves as not knowing their job (aka interns).

**Comment by keronshb** [comment]
> Yes I know this is basically just an ideas page. All I'm saying is that there should be an option for people who have above 24 hours in a department (or whatever number gets settled on in the end) to toggle the intern prefix, instead of forcing the intern prefix to be removed.

24 hours were just an example.

Ideally people can toggle a title on/off regardless of hours. So if someone still wants intern they can slap it on.

**Comment by PJB3005** [comment]
I don't think there should be a significant playtime requirement at all to enable the "teacher" role. The point is that it shouldn't be an indicator of prestige or a flex in and of itself.

**Comment by LordEclipse** [comment]
Mentorhelp whitelist system with teacher job title

**Comment by VasilisThePikachu** [comment]
#22418 should probably also have a solution before removing interns too.

**Comment by OtherSkittle** [comment]
I personally think players with the intern prefix should see a marking (maybe like sechud but in a different location) on players with the teacher prefix so they are easy to find without shift clicking. Maybe have it be a on by default option in whatever part of the ui will handle choosing prefixes so it isn't annoying or immersion breaking or whatever.

There could also maybe be a setting in the PDA to toggle the teacher prefix mid game in case they don't want to teach anymore, are planning to go ssd, or they roll antag and want to do their objectives without being bothered.

**Comment by enumerate0** [comment]
Discussion of this topic has taken place at [this message](https://discord.com/channels/310555209753690112/310555209753690112/1186816283057070181) in Discord.
